### PR TITLE
[Gazebo 9] Added test to check collisions equal to zero

### PR DESCRIPTION
**Pull request by ahcorde**
I missed to add tests in this PR https://github.com/osrf/gazebo/pull/2768. 

I added a world with a box, sphere and cylinder with collisions equal to zero. The test will try to load the world, no segmentation fault should happens

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: gazebo/rendering/Visual_TEST.cc [hunk 1]
```diff
@@ -1588,6 +1588,33 @@ TEST_F(Visual_TEST, ConvertVisualType)
       rendering::Visual::ConvertVisualType(msgs::Visual::PHYSICS));
 }
 
+TEST_F(Visual_TEST, collisionZero)
```
Comment (mabelzhang) [inline gazebo/rendering/Visual_TEST.cc]: Test name should start with capital letter

File: gazebo/rendering/Visual_TEST.cc [hunk 2]
```diff
@@ -1588,6 +1588,33 @@ TEST_F(Visual_TEST, ConvertVisualType)
...
+{
+  // This test checks that there isn't a segmentation fault when inserting zero collision geometries
```
Comment (mabelzhang) [inline gazebo/rendering/Visual_TEST.cc]: Code check is saying lines 1593 and 1594 should be < 80 chars long.

**Review by mabelzhang — COMMENTED** [review]
I tried the test, it passes, but it also passes when I reverted to the original code that produced the GUI bug before the change in https://github.com/osrf/gazebo/pull/2768/files .

I reverted Visual.cc lines 814-823 to the original lines 814-815. In GUI, when I do View > Collisions, I get
```
gzclient: /build/ogre-1.9-B6QkmW/ogre-1.9-1.9.0+dfsg1/OgreMain/src/OgreNode.cpp:630: virtual void Ogre::Node::setScale(const Ogre::Vector3&): Assertion `!inScale.isNaN() && "Invalid vector supplied as parameter"' failed.
Aborted (core dumped)
```
But the test is passing with this original version of code. So it is not exposing the bug.

File: gazebo/rendering/Visual_TEST.cc [hunk 3]
```diff
@@ -1588,6 +1588,33 @@ TEST_F(Visual_TEST, ConvertVisualType)
...
+  rendering::VisualPtr box, sphere, cylinder;
+  while ((!box || !sphere || !cylinder) && sleep < maxSleep)
+  {
+    event::Events::preRender();
+    event::Events::render();
+    event::Events::postRender();
+
+    box = scene->GetVisual("box");
+    cylinder = scene->GetVisual("cylinder");
+    sphere = scene->GetVisual("sphere");
```
Comment (chapulina) [inline gazebo/rendering/Visual_TEST.cc:1615]: Should you check that these aren't `nullptr` after the loop?

**Review by chapulina — COMMENTED** [review]
> I tried the test, it passes, but it also passes when I reverted to the original code

Maybe the failure condition can be triggered by calling `scene->ShowCollisions(true)`.

File: gazebo/rendering/Visual_TEST.cc (hunk 3 above)
Comment (ahcorde) [reply gazebo/rendering/Visual_TEST.cc:1615]: Test is passing:

```
133: [ RUN      ] Visual_TEST.CollisionZero
133: [Msg] Waiting for master.
133: [Msg] Connected to gazebo master @ http://127.0.0.1:11345
133: [Msg] Publicized address: 172.17.0.2
133: [Dbg] [ServerFixture.cc:203] ServerFixture load in 0.6 seconds, timeout after 600 seconds
133: [Err] [Visual.cc:823] box::link::collision__COLLISION_VISUAL__: Size of the collision contains one or several zeros. Collisions may not visualize properly.
133: [Err] [Visual.cc:823] sphere::link::collision__COLLISION_VISUAL__: Size of the collision contains one or several zeros. Collisions may not visualize properly.
133: [Err] [Visual.cc:823] cylinder::link::collision__COLLISION_VISUAL__: Size of the collision contains one or several zeros. Collisions may not visualize properly.
133: [Dbg] [ServerFixture.cc:129] ServerFixture::Unload
133: [       OK ] Visual_TEST.CollisionZero (2810 ms)
133: [ RUN      ] Visual_TEST.Scale
```

**Comment by mabelzhang** [comment]
Verified that now the test does fail with the error above when `Visual.cc` is reverted.
Test passes with the fix.

`tools/code_check.sh` still reporting these though:
```
./gazebo/rendering/Visual_TEST.cc:1593:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
./gazebo/rendering/Visual_TEST.cc:1595:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
./gazebo/rendering/Visual_TEST.cc:1619:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
```

**Comment by ahcorde** [comment]
Thank you @mabelzhang ! I got confused with the 100 characters per line in ROS

**Comment by mabelzhang** [comment]
Yeah I do that too. I don't even remember if ign-gazebo is 80 or 100.
I'll wait for Jenkins to finish.

**Review by mabelzhang — APPROVED** [review]
CI looks fine. Feel free to merge.

**Pull request merged**